pkgs/servers/binserve/default.nix

```
@@ -0,0 +1,31 @@
+{ lib, stdenv, fetchFromGitHub, rustPlatform }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "binserve";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "mufeedvh";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-Chm2xPB0BrLXSZslg9wnbDyHSJRQAvOtpH0Rw6w1q1s=";
+  };
+
+  cargoLock.lockFile = ./Cargo.lock;
+  postPatch = ''
+    cp ${./Cargo.lock} Cargo.lock
+  '';
+
```

I don't believe there are any tests in the source code, so it just builds twice without a good reason.
```suggestion

  doCheck = false;

```